src/vstest.console/Processors/RunSpecificTestsArgumentProcessor.cs

```diff
@@ -315,6 +340,15 @@ private void discoveryRequest_OnDiscoveredTests(object sender, DiscoveredTestsEv
             }
         }
 
+        private void discoveryRequest_OnDiscoveredBucketizedTests(object sender, DiscoveredTestsEventArgs args)
```

Comment:
Why is this needed? Why are we not filtering discovered tests?

Reply:
So the logic for bucketizing was almost identical to test filtering (minus the filtering). So I reused this class. For clarity probably worth creating different classes that can share common logic. I'm happy to clean things up but wanted to make sure you all are fine with the general idea before I spend any more time on this.

Reply:
@vagisha-nidhi does this make sense?